docs/source/notes/cuda.rst

```diff
@@ -12,35 +12,47 @@ However, once a tensor is allocated, you can do operations on it irrespective
 of the selected device, and the results will be always placed in on the same
 device as the tensor.
 
-Cross-GPU operations are not allowed by default, with the only exception of
-:meth:`~torch.Tensor.copy_`. Unless you enable peer-to-peer memory access, any
-attempts to launch ops on tensors spread across different devices will raise an
-error.
+Cross-GPU operations are not allowed by default, with the exception of
+:meth:`~torch.Tensor.copy_` and other methods like copy like
+:meth:`~torch.Tensor.to` and :meth:`~torch.Tensor.cuda`.
+Unless you enable peer-to-peer memory access, any attempts to launch ops on
+tensors spread across different devices will raise an error.
 
 Below you can find a small example showcasing this::
 
-    x = torch.cuda.FloatTensor(1)
-    # x.get_device() == 0
-    y = torch.FloatTensor(1).cuda()
-    # y.get_device() == 0
+    cuda = torch.device('cuda')     # Default CUDA device
+    cuda0 = torch.device('cuda:0')
+    cuda2 = torch.device('cuda:2')  # GPU 2 (these are 0-indexed)
+
+    x = torch.tensor([1., 2.], device=cuda0)
+    # x.device is device(type='cuda', index=0)
+    y = torch.tensor([1., 2.]).cuda()
+    # y.device is device(type='cuda', index=0)
 
     with torch.cuda.device(1):
         # allocates a tensor on GPU 1
-        a = torch.cuda.FloatTensor(1)
+        a = torch.tensor([1., 2.], device=cuda)
 
         # transfers a tensor from CPU to GPU 1
-        b = torch.FloatTensor(1).cuda()
-        # a.get_device() == b.get_device() == 1
+        b = torch.tensor([1., 2.]).cuda()
+        # a.device.index == b.device.index == 1
+
+        # You can also use ``Tensor.to`` to transfer a tensor:
+        b2 = torch.tensor([1., 2.]).to(device=cuda)
+        b.device.index == b2.device.index == 1
 
         c = a + b
-        # c.get_device() == 1
+        # c.device.index == 1
 
         z = x + y
-        # z.get_device() == 0
+        # z.device.index == 0
 
-        # even within a context, you can give a GPU id to the .cuda call
-        d = torch.randn(2).cuda(2)
-        # d.get_device() == 2
+        # even within a context, you can specify the device
+        # (or give a GPU index to the .cuda call)
+        d = torch.randn(2, device=cuda2)
+        e = torch.randn(2).to(device=cuda2)
+        f = torch.randn(2).cuda(2)
+        # d.device.index == e.device.index == f.device.index == 2
```

Comment:
you can also pass a device to cuda; again, I don't see the advantage of jumping back and forth between index and device.